data_structures/trie_tree.cpp

```diff
@@ -1,91 +1,209 @@
-#include <stdbool.h>
-#include <stdio.h>
-
+/**
+ * @file
+ * @author [@Arctic2333](https://github.com/Arctic2333)
+ * @author [Krishna Vedala](https://github.com/kvedala)
+ * @brief Implementation of [Trie](https://en.wikipedia.org/wiki/Trie) data
+ * structure for English alphabets in small characters.
+ * @note the function ::data_structure::trie::deleteString might be erroneous
+ * @see trie_modern.cpp
+ */
+#include <array>
+#include <cassert>
 #include <iostream>
+#include <memory>
 #include <string>
+#include <vector>
 
-// structure definition
-typedef struct trie {
-    struct trie* arr[26];
-    bool isEndofWord;
-} trie;
-
-// create a new node for trie
-trie* createNode() {
-    trie* nn = new trie();
-    for (int i = 0; i < 26; i++) nn->arr[i] = NULL;
-    nn->isEndofWord = false;
-    return nn;
-}
+/** \namespace data_structure
+ * \brief Data-structure algorithms
+ */
+namespace data_structure {
```

Comment:
```suggestion
namespace data_structures {
```

Reply:
in the existing code, it is marked as `data_structure`. Changing it now will create two separate namespaces.